deepmd/pt/train/training.py

```diff
@@ -215,12 +228,26 @@ def single_model_stat(
             _stat_file_path,
             _data_requirement,
             finetune_has_new_type=False,
+            modifier=None,
         ):
             _data_requirement += get_additional_data_requirement(_model)
             _training_data.add_data_requirement(_data_requirement)
             if _validation_data is not None:
                 _validation_data.add_data_requirement(_data_requirement)
 
+            # modify data
+            if modifier is not None:
+                log.info(f"Using {modifier.modifier_type} as data modifier")
+                for _data in [_training_data, _validation_data]:
+                    if _data is not None:
+                        all_sampled = make_stat_input(
+                            _data.systems,
+                            _data.dataloaders,
+                            -1,
+                        )
+                        for sampled in all_sampled:
+                            modifier.modify_data(sampled)
```

Comment:
I noticed that this modification only affects the data statistics. It still uses the original data in `get_data` for actual training.